### PR TITLE
[libpng] Add file usage.txt to describe the usage of port libpng

### PR DESCRIPTION
**Pull request by RookieTars**
**Describe the pull request**

- #### What does your PR fix?  
  Add usage.txt in directory ports/libpng to describe the usage of package libpng in CMake targets.

- #### Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  `Yes`

### CODE REVIEW

File: ports/libpng/usage.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The package libpng is compatible with built-in CMake targets:
+
+    find_package(libpng REQUIRED)
+    target_link_libraries(main PRIVATE ${PNG_LIBRARY})
+    target_include_directories(main PRIVATE ${PNG_INCLUDE_DIR})
```
Comment (Neumann-A) [inline ports/libpng/usage.txt:5]: why? use targets? Isn't the automatic usage not good enough?

there is also cmakes FindPNG which can be used.

Comment (RookieTars) [reply ports/libpng/usage.txt:5]: Do you mean `find_package(PNG REQUIRED)` ? It doesn't work on my machine.

Comment (Osyotr) [reply ports/libpng/usage.txt:5]:
```
find_package(libpng REQUIRED)
target_link_libraries(${target_name} PRIVATE png)
```
This is how it's used in my project. I think this is also what vcpkg prints after install.

**Comment by Hoikas** [comment]
This port seems to already have a usage message
```
The package libpng provides CMake targets:

    find_package(libpng CONFIG REQUIRED)
    target_link_libraries(main PRIVATE png)
```
is this incorrect?

File: ports/libpng/usage.txt (hunk 1 above)
Comment (Neumann-A) [reply ports/libpng/usage.txt:5]: for static builds it is `png_static`

Comment (RookieTars) [reply ports/libpng/usage.txt:5]: @Osyotr You're right.I didn't pay attention to the info after install and couldn't find any information from this repo.

Comment (RookieTars) [reply ports/libpng/usage.txt:5]: @Neumann-A Sorry, I can't understand that. 😢  I'm still very new to CMake and vcpkg. But I found 
```
find_package(PNG REQUIRED)
target_link_libraries(main PRIVATE PNG::PNG)
```
works on my machine. Is that the correct usage of FindPNG?

**Comment by RookieTars** [comment]
@Hoikas That's right! But I couldn't find it.

**Comment by dg0yt** [comment]
The port doesn't have a usage file. The usage information is generated automatically (but seems to work).

I would still recommend the [CMake find module](https://cmake.org/cmake/help/latest/module/FindPNG.html) if you don't want to depend on either vpckg or a recent enough libpng.

~~~cmake
find_package(PNG REQUIRED)
target_link_libraries(main PRIVATE PNG::PNG) # since CMake 3.5
~~~

If this doesn't work, it is either wrong usage of vcpkg, or a bug.

**Comment by dg0yt** [comment]
> ```
>     target_link_libraries(main PRIVATE png)
> ```
> 
> 
>     
>   
> 
> is this incorrect?

As pointed out by @Neumann-A, this is incorrect for static builds. Adding to #20190.

**Comment by dg0yt** [comment]
There is now a usage file based on `PNG`.

**Comment by JackBoosY** [comment]
> There is now a usage file based on `PNG`.

Correct. So I'm going to close this PR since it's invalid.

**Comment by RookieTars** [comment]
> There is now a usage file based on `PNG`.

Appreciate that.

**Pull request closed without merging**